ini/_ini_map.ts

```diff
@@ -302,7 +302,13 @@ export class IniMap<T = any> {
     }
     const reviverFunc: ReviverFunction = typeof reviver === "function"
       ? reviver
-      : (_key, value, _section) => value;
+      : (_key, value, _section) => {
+        if (value === "undefined") return undefined;
+        if (!isNaN(+value) && !value.includes('"')) return parseInt(value);
```

Comment:
Why not just use `!isNaN(value)`?

Reply:
There's no particular reason, I just put it that way out of habit

Reply:
This allows expression like `1e3` as parsed as number, and it results `1`. Is that intentional?

Reply:
> Edit: I just checked and isNaN expects a number, so there's no way to use `!isNaN(value)`

Oh you're right. I forgot we're using typescript not javascript lol

Reply:
If its a number but has a `"` the user wants as a string and not as a number

Reply:
Fair enough